src/bio.c

```diff
@@ -263,6 +264,12 @@ int wolfSSL_BIO_read(WOLFSSL_BIO* bio, void* buf, int len)
         if (ret > 0) {
             sz = ret; /* adjust size for formatting */
         }
+        else {
+            if (wolfSSL_BIO_supports_pending(bio) &&
+                wolfSSL_BIO_ctrl_pending(bio) == 0) {
+                ret = WOLFSSL_CBIO_ERR_WANT_READ;
```

Comment:
Why would we want to return a `WOLFSSL_CBIO_ERR_WANT_READ` from a BIO_READ call? We should be returning 0. This pending check should be only done in the context of a SSL read . From the docs:
>All other functions return either the amount of data successfully read or written (if the return value is positive) or that no data was successfully read or written if the result is 0 or -1

Reply:
So move it into the BioReceive... Will do. Thanks